pkg/trait/telemetry.go

```diff
@@ -84,36 +56,50 @@ var (
 )
 
 // NewTelemetryTrait instance the telemetry trait as a BaseTrait capable to inject quarkus properties.
-func NewTelemetryTrait() trait.Trait {
+func NewTelemetryTrait() Trait {
 	return &telemetryTrait{
-		BaseTrait: trait.NewBaseTrait("telemetry", trait.TraitOrderBeforeControllerCreation),
+		BaseTrait: NewBaseTrait("telemetry", TraitOrderBeforeControllerCreation),
 	}
 }
 
-func (t *telemetryTrait) Configure(e *trait.Environment) (bool, *trait.TraitCondition, error) {
+func (t *telemetryTrait) Configure(e *Environment) (bool, *TraitCondition, error) {
 	if e.Integration == nil || !ptr.Deref(t.Enabled, false) {
 		return false, nil, nil
 	}
 
-	if !ptr.Deref(t.Auto, true) {
-		return true, nil, nil
+	var condition *TraitCondition
+
+	if _, isAddon := e.Integration.Spec.Traits.Addons["telemetry"]; isAddon {
```

Comment:
Ah, we also need to add a `// Deprecated` comment as it's the way I usually scan code for deprecations to be eventually removed.